### PR TITLE
[Draft] Support multi config 

### PR DESCRIPTION
**Pull request by larohra**
## Description
PR for adding support for MultiConfig 

Issue # 

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azurelinuxagent/ga/exthandlers.py [hunk 1]
```diff
@@ -98,10 +100,12 @@ class ValidHandlerStatus(object):
 class ExtCommandEnvVariable(object):
     Prefix = "AZURE_GUEST_AGENT"
     DisableReturnCode = "{0}_DISABLE_CMD_EXIT_CODE".format(Prefix)
+    DisableReturnCodeMultipleExtensions = "{0}_DISABLE_CMD_EXIT_CODES_MULTIPLE_EXTENSIONS".format(Prefix)
```
Comment (narrieta) [inline azurelinuxagent/ga/exthandlers.py:103]: do we need a new variable or can we reuse _DISABLE_CMD_EXIT_CODE?

**Review by narrieta — APPROVED** [review]
minor comments

File: azurelinuxagent/common/protocol/restapi.py [hunk 2]
```diff
@@ -114,6 +130,16 @@ def __init__(self,
         self.dependencyLevel = dependencyLevel
         self.state = state
 
+    def sort_key(self, handler_state):
```
[the remark this answers is not in the document]
Comment (larohra) [reply azurelinuxagent/common/protocol/restapi.py:133]: Sure, I was keeping it consistent with our current code (ExtHandlers object has this too) but renaming it makes more sense.

File: azurelinuxagent/common/protocol/goal_state.py [hunk 3]
```diff
@@ -652,20 +652,21 @@ def __parse_extension_runtime_settings(plugin_settings_node, extension_runtime_s
             dependency_level = ExtensionsConfig.__get_dependency_level_from_node(depends_on_node, extension_name)
             dependency_levels[extension_name] = dependency_level
 
+        ext_handler.supports_multi_config = True
```
[the remark this answers is not in the document]
Comment (larohra) [reply azurelinuxagent/common/protocol/goal_state.py:655]: My idea was to mainly mark it as multi-config after we've made sure there are no more config errors in this. In the reporting too we mainly need the extension name and seq_no for multi-config, if either is missing then reporting would not be correct so I figured I'd just set it down in the function over in the caller. 
I'll re-verify this though

**Comment by larohra** [comment]
Got a rough review.

**Pull request closed without merging**